content/en/blog/2023/native-sidecars/index.md

```diff
@@ -0,0 +1,201 @@
+---
+title: Kubernetes Native Sidecars in Istio
+description: Demoing the new SidecarContainers feature with Istio.
+publishdate: 2023-07-25
+attribution: "John Howard (Google)"
+keywords: [istio,sidecars,kubernetes]
+---
+
+From the beginning of Istio, applications have been augmented with service mesh functionality by using the "sidecar" pattern:
+each Pod gets an additional container running alongside it.
+In Istio's case, this additional container runs the Istio proxy, giving us the classic Istio architecture:
+
+{{< image width="80%"
+    link="/docs/ops/deployment/architecture/arch.svg"
+    caption="Istio Architecture"
+    >}}
+
+While most common in service meshes, the sidecar pattern is used for other contexts as well, such as monitoring agents.
+
+## Sidecar woes
+
+Sidecar containers give a lot of power, but come with some issues.
+While containers within a Pod can share some things, their *lifecycle's* are entirely decoupled.
+To Kubernetes, both of these containers are functionally the same.
+
+However, in Istio they are not the same - the Istio container is required for the primary application container to run,
+and has no value without the primary application container.
```

Comment:
I'd say the opposite sometimes also true - when sidecar provides secure connection, primary container doesn;t make sense without the sidecar as it cannot communicate. Thus we will restart siecar even when Job is finite.

Reply:
True - I guess neither makes sense without the other :-) I was thinking the sidecar does nothing on its own though, it only serves to augment the main app. let me reword this